sdk/samples/iot/provisioning/README.md

```diff
@@ -11,50 +11,129 @@ urlFragment: iot-provisioning-samples
+* Have VCPKG and [Eclipse Paho MQTT C client](https://www.eclipse.org/paho/) installed.  Use the directions [here](https://github.com/Azure/azure-sdk-for-c#development-environment) to download VCPKG and install Paho MQTT.
 
-   Following section document various examples.
+## Getting Started
 
-1. [IoT Provisioning using Paho][samples_paho]: Register a device using the IoT Provisioning client and the [Eclipse Paho MQTT C client][Eclipse_Paho].
+### Paho IoT Provisioning (Certificates)
+This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c) uses x509 authentication to connect to Azure IoT Hub Device Provisioning Service (DPS).  To easily run this sample, we have provided a script to generate a self-signed device certification used for device authentication. This script is intended **for sample use only and not to be used in production code**.
+
+1. Enter the directory `/azure-sdk-for-c/sdk/samples/iot/provisioning/src/` and run the script using the following form:
+
+    Linux:
+    ```bash
+    ./generate_certificate.sh
+    ```
+    Windows:
+    ```cmd
+    generate_certificates.cmd
+    ```
+2. Set the environment variable `AZ_IOT_DEVICE_X509_CERT_PEM_FILE` noted at the bottom of the output.
+3. In your Azure DPS, add a new individual device enrollment using the recently generated `device_ec_cert.pem` file.  See [here](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device-x509#create-a-device-enrollment-entry-in-the-portal) for further instruction.  After creation, the Registration ID of your device should appear as `paho-sample-device1` in the Individual Enrollments tab.
+
+### Paho IoT Provisioning (SAS)
+This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/provisioning/src/paho_iot_provisioning_sas_example.c) uses SAS symmetric key authentication to connect to Azure IoT Hub Device Provisioning Service (DPS). When enrolling a device with the DPS, an option to auto-generate the key will be provided.  
+
+* In your Azure DPS, add a new individual device enrollment using SAS. See [here](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device-symm-key#create-a-device-enrollment-entry-in-the-portal) for further instruction. After creation, the Registration ID of your device will appear in the Individual Enrollments tab.
+
+
+
+## Build and Run the Sample
+
+1. Set the remaining environment variables.  Setting a variable will take the following form:
+
+	  Linux:
+	  ```bash
+	  export ENV_VARIABLE_NAME=VALUE
+	  ```
+	  Windows:
+	  ```cmd
+	  set "ENV_VARIABLE_NAME=VALUE"
+	  ```
+	#### Paho IoT Provisioning (Certificates)
+  * `VCPKG_DEFAULT_TRIPLET` and `VCPKG_ROOT`: These should already be set per these [directions](https://github.com/Azure/azure-sdk-for-c#development-environment).
```

Comment:
relative link?

Reply:
I forgot to bring this up in our meeting, but there are folks on our engsys team working towards fixing potential 404 or incorrect links in READMEs and docs across all the languages, and there recommendation was to use complete links rather than relative ones.

For example, see [this file change](https://github.com/Azure/azure-sdk-for-js/pull/9978/files#diff-f57514279a5dcfcfdf7059f4ec56e4a9R8) from https://github.com/Azure/azure-sdk-for-js/pull/9978 OR https://github.com/Azure/azure-sdk-for-python/pull/12543
cc @KarishmaGhiya

Reply:
i see.  ok thanks for the info ahson!